### PR TITLE
9 j unit for every java method

### PR DESCRIPTION
**Pull request by jprietot**

### CODE REVIEW

File: carAPI/car.log [hunk 1]
```diff
@@ -0,0 +1,135 @@
+[2021-04-09 12:40:58] [ INFO ] [CarResource:46] Getting cars list
```
Comment (rafapaezbas) [inline carAPI/car.log:1]: Maybe you should include this file in the .gitignore

File: carAPI/src/test/java/boundary/CarResourceTests.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		car.setId(3);
+		car.setBrand("TestCar");
+		car.setCountry("Spain");
+		car.setRegistration(LocalDateTime.now());
+		car.setCreatedAt(LocalDateTime.now());
+		car.setLastUpdated(LocalDateTime.now());
+	}
+	
+	@Test
+	public void testGetCars() {
```
Comment (rafapaezbas) [inline carAPI/src/test/java/boundary/CarResourceTests.java:59]: Tests are looking good, also keep in mind:
Unit tests goal is to test the most significant use cases that a single method can handle. That means, try to explorer also the weird use cases in the tests. For instance, what if the CarService instead of returning a proper list of cars, returns a null. Having unit tests for corner cases is a good idea from my point of view. Always try to tests unexpected behaviours!

File: carAPI/src/test/java/boundary/CarResourceTests.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		List<Car> carList = new ArrayList<Car>();
+		when(this.carService.getCars()).thenReturn(carList);
+		Response response = this.carResource.getCars();
+		assertEquals(Status.OK.getStatusCode(), response.getStatus());
+	}
+	
+	@Test
+	public void testGetCar() {
```
Comment (rafapaezbas) [inline carAPI/src/test/java/boundary/CarResourceTests.java:67]: Test also the negative case! CarService doesnt return any car, or the input of the resource is null.

File: carAPI/src/test/java/boundary/CarResourceTests.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	@InjectMocks
+	private CarResource carResource;
+	
+	@Mock
+	private CarService carService;
+	
+	@Mock
+	private UriInfo uriInfo;
+	
+	private Car car;
```
Comment (rafapaezbas) [inline carAPI/src/test/java/boundary/CarResourceTests.java:45]: you can make this variable final, this way you cannot change the car properties n any test having side effects for other tests. Final variables (a.k.a constants) are usually a good practice in programming.
